### PR TITLE
Initial release notes for computed and substates

### PR DESCRIPTION
**Pull request by alice-i-cecile**
This is an unusually complex feature: please complain if you can't follow the example or don't understand the point of the feature!

### CODE REVIEW

**Comment by lee-orr** [comment]
This is an awesome write-up! My only question is whether we should mention the two potentially breaking changes:
- if you are using `default-features = false`, you need to manually enable the feature / if using `bevy_ecs` directly you need to import `bevy_states`
- if you had multiple places where you triggered `apply_state_transition`, you should now instead run the `StateTransition` schedule

File: release-content/0.14/release-notes/11426_Computed_State__Sub_States.md [hunk 1]
```diff
@@ -1 +1,107 @@
...
+App::new()
+   .init_state::<GameState>()
+   .add_computed_state::<InGame>()
+   .add_substate::<GamePhase>()
+```
+
+Just like any other state, computed states and substates work with all of the tools you're used to:
+the `State` and `NextState` resources, `OnEnter`, `OnExit` and `OnTransition` schedules and the `in_state` run condition.
+
+The only exception is that, for correctness reasons, computed states *cannot* be mutated manually.
```
Comment (MiniaczQ) [inline release-content/0.14/release-notes/11426_Computed_State__Sub_States.md:101]:
```suggestion
The only exception is that, for correctness, computed states *cannot* be mutated through `NextState`.
```
Bit more direct?

File: release-content/0.14/release-notes/11426_Computed_State__Sub_States.md [hunk 2]
```diff
@@ -1 +1,107 @@
...
+Instead, they are strictly derived from their input states, and added, removed and updated automatically during state transitions based on the provided `compute` method.
+
+All of Bevy's state tools are now found in a dedicated `bevy_states` crate, which can be controlled via a feature flag.
```
Comment (MiniaczQ) [inline release-content/0.14/release-notes/11426_Computed_State__Sub_States.md:104]: Note about a breaking change here?

**Review by MiniaczQ — APPROVED** [review]
Looks correct and inviting

**Comment by alice-i-cecile** [comment]
I think the breaking change is best relegated to the migration guide: using both pure bevy_ecs and still using states is a minority of a minority situation.

File: release-content/0.14/release-notes/11426_Computed_State__Sub_States.md [hunk 3]
```diff
@@ -1 +1,107 @@
...
+    }
+}
+
+fn open_menu() {
+    // Show the settings menu...
+}
+
+App::new()
+    .init_state::<GameState>()
+    .add_systems(Update, handle_escape_pressed.run_if(in_state(GameState::Menu)));
```
Comment (theotherphil) [inline release-content/0.14/release-notes/11426_Computed_State__Sub_States.md:28]: Should this say `in_state(GameState::InGame)` or do I just not understand Bevy states?

Comment (alice-i-cecile) [reply release-content/0.14/release-notes/11426_Computed_State__Sub_States.md:28]: Yes, it should.

File: release-content/0.14/release-notes/11426_Computed_State__Sub_States.md [hunk 4]
```diff
@@ -1 +1,107 @@
...
+```rust
+/// Let's begin with our simple state above
+#[derive(States, Clone, PartialEq, Eq, Hash, Debug, Default)]
+enum GameState {
+    #[default]
+    Menu,
+    // The addition of this field means that simply checking for `GameState::InGame` doesn't work:
+    // the values are different based on its value and we may want to distinguish between game systems that run
+    // when the game is paused or not!
+    InGame {
```
Comment (B-Reif) [inline release-content/0.14/release-notes/11426_Computed_State__Sub_States.md:49]: I think it's a little confusing to have this enum case share a name with our derived state example. Maybe we could rename this to `Running` or something?

**Review by B-Reif — COMMENTED** [review]
Looks good, a couple things I don't understand.

**Pull request merged**